registry/storage/filewriter_test.go

```diff
@@ -165,8 +165,9 @@ func TestSimpleWrite(t *testing.T) {
 
 func TestBufferedFileWriter(t *testing.T) {
 	ctx := context.Background()
-	writer, err := newFileWriter(ctx, inmemory.New(), "/random")
 
+	writerp, err := newFileWriter(ctx, inmemory.New(), "/random")
+	writer := *writerp
```

Comment:
Of course you could argue that bufferedFileWriter is not designed to be dereferenced and used in the way I do in the test case, but it that case the dereference at https://github.com/aibaars/distribution/blob/9f83bfc071094019b942c4332d17327b4dc5cdfc/registry/storage/linkedblobstore.go#L200 should be fixed instead. 

I opted for fixing the problem in bufferedFileWriter, because I think it is not nice to have a type that has subtle misbehaviour when used by value instead of by reference.